user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/UserServiceAsyncProxyErrorTest.java

```diff
@@ -0,0 +1,196 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.component.user.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.cache.CacheAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.redis.RedisAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.serializer.support.SerializationFailedException;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.support.TestPropertySourceUtils;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.component.user.service.UserServiceAsyncProxyErrorTest.RedisInitializer;
+import uk.gov.gchq.palisade.service.user.config.ApplicationConfiguration;
+import uk.gov.gchq.palisade.service.user.exception.NoSuchUserIdException;
+import uk.gov.gchq.palisade.service.user.model.AuditableUserResponse;
+import uk.gov.gchq.palisade.service.user.model.UserRequest;
+import uk.gov.gchq.palisade.service.user.service.UserServiceAsyncProxy;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests to verify the handling of exceptions, and the population of audit objects during stream processing
+ */
+@SpringBootTest(
+        classes = {ApplicationConfiguration.class, CacheAutoConfiguration.class},
+        webEnvironment = WebEnvironment.NONE,
+        properties = {"spring.cache.redis.timeToLive=1s"}
+)
+@EnableCaching
+@ContextConfiguration(initializers = {RedisInitializer.class})
+@Import(RedisAutoConfiguration.class)
+@ActiveProfiles({"redis"})
+class UserServiceAsyncProxyErrorTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserServiceAsyncProxyErrorTest.class);
+
+
+    private static final Function<Integer, String> REQUEST_FACTORY_JSON = i ->
```

Comment:
I get why we'd have this in contract tests, but why here?

Reply:
This isn't actually used properly as we return the same json String each time. Adjusted to just be a `String` value.